framework/src/main/java/org/checkerframework/framework/type/typeannotator/DefaultForTypeAnnotator.java

```diff
@@ -193,4 +212,151 @@ public DefaultForTypeAnnotator addStandardDefaults() {
 
         return this;
     }
+
+    /**
+     * Apply defaults based on a variable name to a type.
+     *
+     * @param type a type to apply defaults to
+     * @param name the name of the variable that has type {@code type}, or the name of the method
+     *     whose return type is {@code type}
+     */
+    public void defaultTypeFromName(AnnotatedTypeMirror type, String name) {
+        // TODO: Check whether the annotation is applicable to this Java type?
+        AnnotationMirror defaultAnno = namesRegexes.getDefaultAnno(name);
+        if (defaultAnno != null) {
+            if (typeFactory
+                            .getQualifierHierarchy()
+                            .findAnnotationInHierarchy(type.getAnnotations(), defaultAnno)
+                    == null) {
+                type.addAnnotation(defaultAnno);
+            }
+        }
+    }
+
+    @Override
+    public Void visitExecutable(AnnotatedExecutableType type, Void aVoid) {
+        ExecutableElement element = type.getElement();
+
+        Iterator<AnnotatedTypeMirror> paramTypes = type.getParameterTypes().iterator();
+        for (VariableElement paramElt : element.getParameters()) {
+            String paramName = paramElt.getSimpleName().toString();
```

Comment:
Parameters from elements only have names if `-g` is used to compile.  See failing test case I added.

Reply:
I clarified when parameter names have an effect and expanded the test case.